common/json.go

```diff
@@ -104,7 +104,12 @@ func (j *JSONHTTPClient) Delete(ctx context.Context, url string, headers ...Head
 func parseJSONResponse(res *http.Response, body []byte) (*JSONHTTPResponse, error) {
 	// empty response body should not be parsed as JSON since it will cause ajson to err
 	if len(body) == 0 {
-		return nil, nil //nolint:nilnil
+		return &JSONHTTPResponse{
```

Comment:
It caused a panic during testing. It can happen in prod. This was something preserved during refactoring but not addressed.
At least the caller should check if Body is nil. Sometimes GET can return nothing, shouldn't make assumptions.